src/main/java/com/mraof/minestuck/item/weapon/AspectBasedEffectWeaponItem.java

```diff
@@ -41,18 +33,25 @@ public AspectBasedEffectWeaponItem(IItemTier tier, int attackDamageIn, float att
     }
     
     @Override
-    public void inventoryTick(ItemStack stack, World worldIn, Entity entityIn, int itemSlot, boolean isSelected)
+    public boolean hitEntity(ItemStack itemStack, LivingEntity target, LivingEntity attacker)
     {
-        if(isSelected && entityIn instanceof ServerPlayerEntity)
+        Title title = PlayerSavedData.getData((ServerPlayerEntity) attacker).getTitle();
+        
+        if(attacker instanceof ServerPlayerEntity && attacker.getRNG().nextFloat() < .1)
         {
-            Title title = PlayerSavedData.getData((ServerPlayerEntity) entityIn).getTitle();
             if(title != null)
             {
                 if (title.getHeroAspect() == aspect)
                 {
-                    ((ServerPlayerEntity) entityIn).addPotionEffect(new EffectInstance(effect.get(), duration, effectTier));
+                    ((ServerPlayerEntity) attacker).addPotionEffect(new EffectInstance(effect.get(), duration, effectTier));
                 }
             }
         }
+    
+        if (itemStack.getItem() == MSItems.CLOWN_CLUB) {
```

Comment:
A non-hardcoded method is preferred over this, but you can wait with that until someone tries to implement a modular system for weapon effects

Reply:
Understood, I felt it was unnecessary to create a new class just for the Clown Club to be able to honk